File: Go/matrices/baseMatrix.go
package matrices

import "os"

// Matrix interface.
type BaseMatrix interface {
	// File input.
	In(line string)

	// Random input.
	RandomIn()

	// File output
	Out(f *os.File)

	// Returns average of all elements of matrix.
	GetAverage() float64
}

File: Rust/src/matrices/base_matrix.rs
use std::fs::File;

pub trait BaseMatrix: Sized {
    fn input(&mut self, info: &mut [String], index: &mut usize);
    fn random_input(&mut self);
    fn output(&self, file: &mut File) -> Result<(), std::io::Error>;
    fn get_average(&self) -> f64;
}
